contributors/design-proposals/service-injection-policy.md

```diff
@@ -0,0 +1,444 @@
+# Service Injection Policy
+
+  * [Abstract](#abstract)
+  * [Motivation](#motivation)
+  * [Constraints and Assumptions](#constraints-and-assumptions)
+  * [Use Cases](#use-cases)
+    * [Loose coupling between services and their consumers](#loose-coupling-between-services-and-their-consumers)
+  * [Proposed Changes](#proposed-changes)
+    * [ServiceInjectionPolicy API object](#serviceinjectionpolicy-api-object)
+      * [Validations](#validations)
+    * [AdmissionControl Plug-in: ServiceInjectionPolicy](#admissioncontrol-plug-in-serviceinjectionpolicy)
+      * [Behavior](#behavior)
+      * [Optional Behavior with implicit injections](#optional-behavior-with-implicit-injections)
+  * [Examples](#examples)
+    * [Simple Pod Spec Example](#simple-pod-spec-example)
+    * [Pod Spec with `ConfigMap` Example](#pod-spec-with-`configmap`-example)
+    * [ReplicaSet with Pod Spec Example](#replicaset-with-pod-spec-example)
+
+
+## Abstract
+
+Describes a policy resource that allows loose coupling from services to the pods
+that consume them.
```

Comment:
Given these are now PIPs it might be better to describe this in a less "service specific" way. Perhaps:
```
Describes a policy resource that allows for the the loose coupling of a Pod's
definition from additional runtime requirements for that Pod. For example,
mounting of Secrets, or setting additional environment variables,
may not be known at Pod deployment time, but may be required at Pod creation
time.
```